cli/azd/internal/repository/infra_confirm.go

```diff
@@ -61,6 +64,10 @@ func (i *Initializer) infraSpecFromDetect(
 		}
 	}
 
+	if hasDb {
+		spec.KeyVault = &scaffold.KeyVault{}
```

Comment:
This is to bring simplified-init parity with native compose db+key-vault explicit resources in azure.yaml?

Reply:
It's to fix a small regression for `azd init` with `azd config set alpha.compose off` I discovered.

The databases always required a keyvault, but when we shifted the dependency ordering for composability -- the simplified init path which depends on infra generation layer directly broke. This is the minimal fix for this and hoping this part is just left unchanged until we remove the legacy codepath sometime in the future.

Noting that the regression was caught by the tests, but the tests weren't running in CI which was unfortunate.

Reply:
`cd cli/azd/internal/scaffold && go test` should reproduce the failures.

The tests will now run in CI, so at least we'll get coverage...